extensions/native/compiler/src/ir/collections.rs

```diff
@@ -78,10 +78,9 @@ impl<C: Config, V: MemVariable<C>> Array<C, V> {
                 }
             }
             Self::Dyn(ptr, len) => {
-                assert_eq!(V::size_of(), 1, "only support variables of size 1");
                 let len = RVar::from(len.clone());
                 let shift = shift.into();
-                let new_ptr = builder.eval(*ptr + shift);
+                let new_ptr = builder.eval(*ptr + shift * RVar::from(V::size_of()));
```

Comment:
Is our symbolic expression builder smart enough to know that if an `RVar` is `RVar::Const(C::N::ONE)`, then it should not do a multiplication here? I'm worried that this is introducing some cycles/adds.

(I do believe we shouldn't have to worry about it here though, and the logic belongs in the `Mul` implementation for `RVar` or `SymbolicVar`.)

Reply:
It looks like @nyunyunyunyu made this change in `SymbolicVar` so this should not incur any extra cost.